### ISSUE TITLE
When an unfocused, disabled UIStaffSlot loses focus again, it blinks on then off - WotC Bug WITH SOLUTION

### ISSUE THREAD

**Issue by kdm2k6** [long-war-2/lwotc#1014]
**PROBLEM** : When an unfocused, disabled UIStaffSlot loses focus again, it blinks on then off. This is a War of the Chosen Bug !

**SOLUTION** : Follow the precedent set in UIPanel. Only call OnLoseFocus() if the staff slot is focused, and only call OnReceiveFocus() if the staff slot is not focused.

1.] UIStaffSlot --> OnLoseFocus() should become the following :

```
if (bIsFocused)
{
	bIsFocused = false;
	if (`ISCONTROLLERACTIVE)
	{
		MC.FunctionVoid("mouseOut");
	}
}
```

2.] UIStaffSlot --> OnReceiveFocus() should become the following :

```
if (!bIsFocused)
{
	bIsFocused = true;
	if (`ISCONTROLLERACTIVE)
	{
		MC.FunctionVoid("mouseIn");
	}
}
```

**EXPLANATION** : Each time OnLoseFocus() is called on a UIStaffSlot, it calls MC.FunctionVoid("mouseOut"), regardless of bIsFocused; in other words, even if a staff slot has already lost focus, it still loses focus all over again. 

I traced the issue to an ActionScript call : this.realizeState(), which would then call Utilities.RealizeBGState(this.bg,this.isDisabled,this.isFocus,this.mouseDown). The function RealizeBGState() determines the state of the flash element, such as its disabled status and focus status, then calls the function gotoAndPlay().

**Here is where the problem is** :
1.] A focused, disabled staff slot performs : gotoAndPlay("_DisabledOver"). Now, when I sent in the command : MC.ChildFunctionString("bg", "gotoAndStop", "_DisabledOver"), I ended up with a highlighted button.
2.] An unfocused, disabled staff slot performs : gotoAndPlay("_DisabledUp"). Now, when I sent in the command : MC.ChildFunctionString("bg", "gotoAndStop", "_DisabledUp"), I also ended up with a highlighted button.
3.] When I sent in the command : MC.ChildFunctionString("bg", "gotoAndPlay", "_DisabledUp"), I ended up with a flashing button which went : highlighted --> unhighlighted.
4.] **The conclusion** is that gotoAndPlay("_DisabledUp") goes through a highlighted state, possibly but not necessarily _DisabledOver, then finishes in an unhighlighted state.

**How to Reproduce with a controller** : Go to any facility with a disabled staff slot, then press the right or left bumper to access the Avenger shortcuts. Each time you hit the left or right bumper, the disabled staff slot blinks.

This is because every time you hit the left or right bumper, the UIFacility calls ClearFocusOffofFacilityButtons(). From here :
1.] The UIStaffContainer calls OnLoseFocus(), which then calls OnLoseFocus() on all of its staff slots.
2.] ClearFocusOffofFacilityButtons(), itself, calls OnLoseFocus() on all of the staff slots.

Basically, it is guaranteed that OnLoseFocus() is being called on unfocused staff slots.
